### DISCUSSION TITLE
deepin minimal

### DISCUSSION THREAD

**Discussion by antyung88** [linuxdeepin/developer-center#3269] [Developer Discussion | 开发者讨论]
Is there a deepin minimal image available designed for deployment across at scale? At the moment it seems like only desktop version is available...

Would be great if there is a minimal image available with at least mkdir, wget, curl and unzip to install packages to a standard server environment for interactive use!

**Comment by justforlxz** [comment]
We don't provide this environment, but you can create one using pbuilder.

```shell
sudo pbuilder create --architecture amd64 \
                     --mirror "https://mirrors.ustc.edu.cn/deepin" \
                     --distribution apricot \
                     --basetgz ~/apricot-base-amd64.tgz \
                     --allow-untrusted \
                     --debootstrapopts --keyring=/usr/share/keyrings/deepin-archive-camel-keyring.gpg \
                     --debootstrapopts --include=deepin-keyring,ca-certificates
```

A tgz file will be saved in your home directory. This is a compressed file of rootfs, maybe you can complete the construction of the base image with dockerfile. If you want to add some packages, you can append the package at the end with ` --include=deepin-keyring,ca-certificates`.

**Comment by antyung88** [comment]
Seems like debootstrap fails with the latest deepin build... current work around is to use multistrap

```
W: Failure trying to run: chroot "/var/cache/pbuilder/build/561" dpkg --force-overwrite --force-confold --skip-same-version --install
W: See /var/cache/pbuilder/build/561/debootstrap/debootstrap.log for details
E: debootstrap failed
```

**Reply by BLumia** [reply]
`W: See /var/cache/pbuilder/build/561/debootstrap/debootstrap.log for details` may be worth to check the details of the error log.

**Comment by antyung88** [comment]
https://github.com/antyung88/docker-deepin-apricot-minimal
